app/controllers/follows_controller.rb

```diff
@@ -79,16 +79,22 @@ def create
     render json: { outcome: @result }
   end
 
-  def update
-    @follow = Follow.find(params[:id])
-    authorize @follow
-    redirect_to "/dashboard/following" if @follow.update(follow_params)
+  def bulk_update
+    @follows = Follow.where(id: params_for_update.keys).includes(:follower, :followable)
+    authorize @follows
+    redirect_to "/dashboard/following" if @follows.all? { |follow| follow.update(params_for_update[follow.id.to_s]) }
```

Comment:
I wonder what would happen if any of this updates would fail. Should we separate the update step from the `if` and the redirection and do them one at the time?

ps. you can replace `/dashboard/following` with `dashboard_following_path`

Reply:
Oops, thanks for reminding this. I think it would better to add a transaction for this and replace `update` with `update!`, so it will abort the entire update if any update step fails, so the process will be easy to trace. Thanks!